code/modules/mob/dead/new_player/latejoin_menu.dm

```diff
@@ -0,0 +1,161 @@
+#define JOB_CHOICE_YES "Yes"
+#define JOB_CHOICE_REROLL "Reroll"
+#define JOB_CHOICE_CANCEL "Cancel"
+
+/datum/latejoin_menu
+	var/mob/dead/new_player/owner
+
+/datum/latejoin_menu/New(/mob/dead/new_player/owner)
```

Comment:
This should only need to be a singleton with `GLOBAL_DATUM_INIT`, I don't think this needs to be per new player